src/app/report/report-alert-message/report-alert-message.component.ts

```diff
@@ -0,0 +1,34 @@
+import { Component, Input, OnChanges } from '@angular/core';
+import { ReportUtil } from '../../shared/util/report-util';
+import { Report } from '../../shared/interfaces/report';
+import { Checkpoint } from '../../shared/interfaces/checkpoint';
+
+@Component({
+  selector: 'app-report-alert-message',
+  standalone: true,
+  templateUrl: './report-alert-message.component.html',
+  styleUrl: './report-alert-message.component.css',
+})
+export class ReportAlertMessageComponent implements OnChanges {
+  protected readonly ReportUtil = ReportUtil;
+
+  protected anyMessagesPresent: boolean = false;
+  @Input({ required: true }) report!: Report | Checkpoint;
+
+  ngOnChanges(): void {
+    this.checkIfAnyMessagesPresent();
+  }
+
+  checkIfAnyMessagesPresent(): void {
+    if (ReportUtil.isCheckPoint(this.report)) {
+      this.anyMessagesPresent = !!(
+        this.report.streaming ||
+        this.report.stubbed ||
+        !this.report.message ||
+        this.report.encoding ||
+        (this.report.preTruncatedMessageLength && this.report.preTruncatedMessageLength > 0) ||
+        this.report.stubNotFound
+      );
```

Comment:
`!this.report.message` why is this turned negative? Wouldn't not having a message do the exact opposite of what you want?

Reply:
Yeah the property names are a little confusing, this method sets a boolean to true if any alert messages should be shown. One of those alert messages is that the message of a report is null, thats why it checks if its null. I will change the name of the method.

Reply:
Maybe also rename `anyMessagesPresent` to `anyAlertMessagesPresent`

Reply:
Yeah good point, fixed it.